### PR TITLE
middleware: don't use middleware when loading images

### PR DESCRIPTION
**Pull request by JoeGruffins**
check url for gets directed at "/assets" and "/captchas" and ignore them
in the middleware

before this change, open home, fmt.Println("ApplyDbMap") in middleware.go func ApplyDbMap
before

after this change, open home with the same print in the same place
after
I don't believe middleware should be applied to gets to pics for performance reasons, as they aren't using the information, I don't think

### CODE REVIEW

**Comment by JoeGruffins** [comment]
Actually, I think I have a better answer as soon as I figure out the correct negated regex expression.

**Comment by jholdstock** [comment]
Maintaining a regex here is not appropriate because it will require changes every time the application routes change, which is going to be very error-prone and non-intuitive for new devs.

This can be achieved by simply creating a new sub-router for static files, and don't attach any middleware to that subrouter. 

I actually implemented this change some time ago, but it seems like it never made it into master. It must be lost on my local machine somewhere, I will try to find it.

**Comment by JoeGruffins** [comment]
ok! That's what I'm trying to do, but then the regex wouldn't let me do not, and I'm not sure how to specify the folders easily otherwise, yet.

**Comment by jholdstock** [comment]
https://github.com/decred/dcrstakepool/compare/master...jholdstock:routers?expand=1

Found it - see the changes here. I think this achieves the desired result.

**Comment by jholdstock** [comment]
Note that `/robots.txt` and `/favicon.ico` are not required, they do not actually exist at these locations

**Comment by JoeGruffins** [comment]
thanks, I just needed to change the order and goji does the rest. Ill remove those paths then?

File: server.go [hunk 1]
```diff
@@ -246,6 +246,8 @@ func runMain() error {
 	html.Get("/logout", application.Route(controller, "Logout"))
 
 	app.Handle("/api/*", api)
+	app.Handle("/assets/*", static)
```
Comment (jholdstock) [inline server.go]: Adding the `/assets` endpoint to the `app` router will associate all of the middleware with that endpoint.

See how I used a `parent` router with no middleware to solve this problem

Comment (JoeGruffins) [reply server.go]: Did I get it this time?

**Comment by isuldor** [comment]
This is a good improvement.  I just noticed dcrdata defines an nginx `location` to serve static resources directly:

```
# Serve static resources directly.
location ~ ^/(css|js|images|fonts|dist|index.js) {
        gzip_static on; # use .gz files for pre-compressed data
        root /opt/dcrdata/public;
        # Set the Cache-Control and Expires headers for the static assets.
        expires modified 2d;
}
```

**Comment by chappjc** [comment]
That's a solution I would encourage, especially since nginx is so effective at handling cached static resources.

**Comment by JoeGruffins** [comment]
like here ? #433

**Comment by chappjc** [comment]
Oh, good stuff.  ;)

**Comment by JoeGruffins** [comment]
I think #433 is missing an opening bracket.

**Comment by JoeGruffins** [comment]
hey, thanks guyz, quick question. @chappjc  @isuldor 

So with this in the nginx conf:

        # Serve static resources directly.
        location /assets/*
                gzip_static on; # use .gz files for pre-compressed data
                root /opt/dcrstakepool/public;
                # Set the Cache-Control and Expires headers for the static assets.
                expires modified 2d;
        }

Should this pr we are commenting on now not be necessary? I can't tell that the above makes any difference. Even adding the bracket after location /assets/*

**Comment by JoeGruffins** [comment]
Actually, I think I get it now.  Thank you!

**Pull request merged**